examples/ex42.cpp

```diff
@@ -996,8 +1080,9 @@ Apply3D(const int ne, const Array<double> &B_, const Array<double> &G_,
          }
       }
       MFEM_SYNC_THREAD;
+
       auto F = Reshape(&force(0, 0, 0, 0, e), d1d, d1d, d1d, dim);
-      KernelHelpers::CalcGradTSum<dim, d1d, q1d>(B, G, invJ_sigma_detJw, F);
+      KernelHelpers::CalcGradTSum<dim, d1d, q1d>(B, G, smem, invJ_sigma_detJw, F);
```

Comment:
Is it necessary to explicitly specify the template arguments here? I would have guessed the compiler would deduce those values if instead we wrote:

`KernelHelpers::CalcGradTSum(B, G, smem, invJ_sigma_detJw, F);`

https://godbolt.org/z/53arKb1xo

Reply:
`done`